### PR TITLE
Basic RegoRule Operator

### PR DESCRIPTION
**Pull request by lfrancke**
This adds a very basic RegoRule Operator (and CRD).

It has a whole bunch of warts and issues but should serve just fine as a basis to start from.

### CODE REVIEW

**Comment by lfrancke** [comment]
This operator should be renamed to RegoOperator or RegoPolicyOperator I guess

File: operator/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+use std::future::Future;
+use std::path::Path;
+use std::time::{SystemTime, UNIX_EPOCH};
+use tar::{Builder, Header};
+use tracing::{info, trace, warn};
+use warp::Filter;
+
+fn rebuild_bundle(reader: &Store<RegoRule>) -> Result<(), error::Error> {
+    // This will be the timestamp for all files in our bundle
+    let time = SystemTime::now()
```
Comment (soenkeliebau) [inline operator/src/lib.rs:24]: Would it potentially make sense to sync this with the timestamp on the Kubernetes resource that triggered this rebuild? I can see both being useful..

**Review by soenkeliebau — CHANGES_REQUESTED** [review]
Left a few comments, mostly minor stuff, just the service bit needs fixing I think.

File: operator/src/lib.rs (hunk 1 above)
Comment (lfrancke) [reply operator/src/lib.rs:24]: That seems like a sensible idea, yes.
I'm with you. Both have their merits. In reality it probably doesn't matter much.

File: server/src/bin/generate_crd.rs [hunk 2]
```diff
@@ -0,0 +1,7 @@
+use stackable_regorule_crd::RegoRule;
+use std::error::Error;
+
+fn main() -> Result<(), Box<dyn Error>> {
```
[the remark this answers is not in the document]
Comment (lfrancke) [reply server/src/bin/generate_crd.rs:4]: See https://github.com/clux/kube-rs/issues/497 which would make this much easier. We could move the common functionality to the operator-rs

For now I'll leave it like this as it is more flexible.

File: operator/src/lib.rs (hunk 1 above)
Comment (lfrancke) [reply operator/src/lib.rs:24]: Turns out: There is no super trivial "last modified" timestamp.
https://github.com/kubernetes/kubernetes/issues/12430
Server-side apply added timestamps on a per-manager basis but that'd be a lot of extra logic.
In short: I'll keep it as is for now.

File: server/src/bin/generate_crd.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (soenkeliebau) [reply server/src/bin/generate_crd.rs:4]: fine for me

**Comment by soenkeliebau** [comment]
LGTM

**Review by soenkeliebau — APPROVED** [review]

**Pull request merged**